### PR TITLE
Remove references to db postgres and db user postgres

### LINKED ISSUE #223: Postgres user and database should not be used
**Detailed Description**
`postgres` user and database name should not be in the README or defaults (config.json, etc) anywhere in the project.

This could cause people to accidentally add data/schema to this database, which it shouldn't be used for (it's a standard empty db used for connecting).

### PR DESCRIPTION
**Pull request by mike-burrage-hedera**
**Detailed description**:

- Remove all references to database name and user `postgres`
- Pick standard name for db and users and update README, env.sample, config.json.sample
- Fix postgresInit.sql so that it properly creates (a non-`postgres`) database and user. Previously, it would incorrectly set the password. That user specified is the DB owner.

**Which issue(s) this PR fixes**:
Fixes #223 

**Checklist**
- [X] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: src/main/resources/postgres/postgresInit.sql [hunk 1]
```diff
@@ -1,26 +1,15 @@
...
---owner of the database (usually postgres)
-\set db_owner 'postgres'
+\set db_password 'mirror_node_pass'
+\set db_owner 'mirror_node'
 
-CREATE DATABASE :db_name
-    WITH
-    OWNER = :db_owner
-    CONNECTION LIMIT = -1;
+create user :db_user with login createrole password :'db_password';
```
Comment (steven-sheehy) [inline src/main/resources/postgres/postgresInit.sql]: Is this syntax correct with the colon on the outside of the string?

Comment (mike-burrage-hedera) [reply src/main/resources/postgres/postgresInit.sql]: Yes - the old syntax was incorrect and 1 reason why the script didn't work.

**Comment by mike-burrage-hedera** [comment]
We've got 2 different setups right now. 1 postgres 1 hederamirror. I say yes, lets move them to the same naming (which,  with this change, matches the jar and systemd service). Everything's mirror_node/mirror-node.

**Comment by calvinchengx** [comment]
If I am not wrong, there needs to be a grant statement too.

```
GRANT ALL PRIVILEGES ON DATABASE mirror_node to mirror_node;
```

Otherwise, flyway will also fail.

**Comment by mike-burrage-hedera** [comment]
@calvinchengx Is it failing for you?

postgresInit.sql now makes the user the owner of the DB, so it doesn't need any grants.

If you run that script as a createdb/createrole-permissioned account on the database (ie `postgres` user or your default user) it creates the distinct `mirror_node` user and `mirror_node` database owned by that user. No grants required. flyway will take care of the API user creation and those grants (for now until #140 goes in)

**Pull request merged**